xbmc/cores/VideoPlayer/DVDSubtitles/DVDSubtitleParser.h

```diff
@@ -53,14 +54,20 @@ class CDVDSubtitleParserText
      : public CDVDSubtitleParserCollection
 {
 public:
-  CDVDSubtitleParserText(std::unique_ptr<CDVDSubtitleStream> && stream, const std::string& filename)
-    : CDVDSubtitleParserCollection(filename)
-		, m_pStream(std::move(stream))
+  CDVDSubtitleParserText(std::unique_ptr<CDVDSubtitleStream>&& stream,
+                         const std::string& filename,
+                         const char* name)
+    : CDVDSubtitleParserCollection(filename), m_pStream(std::move(stream)), parserName(name)
   {
   }
 
   ~CDVDSubtitleParserText() override = default;
 
+  /*
+   * \brief Returns parser name
+   */
+  const char* GetName() override { return parserName.c_str(); };
```

Comment:
I think this is better as a std::string, you even store it internally with this type. Afaics you only added this for logging purposes, the clog accepts std::string just fine

Reply:
Also no need to move it on construction this is always const, can be declared and initialized on the header file

Reply:
> Also no need to move it on construction this is always const, can be declared and initialized on the header file

is done in the same way of all `DVDOverlayCodecXY` already implemented on original kodi sources
why you want change it all again?
seems functional to me

Reply:
i think i have not userstand exaclty
can you provide a example?

Reply:
You can keep this as is, it doesn't matter much anyway.